components/calendly_v2/sources/invitee-action/invitee-action.mjs

```diff
@@ -0,0 +1,156 @@
+      // "https://api.calendly.com/webhook_subscriptions/69a2dfaa-1a27-4930-b81b-1c7ac0ff6d7b" to "69a2dfaa-1a27-4930-b81b-1c7ac0ff6d7b"
+      const splited = webhookUri.split("/");
+      return splited[splited.length - 1];
+    },
+    async _deactivateWebhook(webhook) {
+      console.log("Deleting webhook...");
+      const res = await this.app.deleteWebhookSubscription(
+        this._getWebhookUuid(webhook.uri),
+      );
+      console.log("Deleted webhook");
+      return res;
+    },
+    async _registerWebhook() {
+      if (!this.events || this.events.length === 0) {
+        throw new Error("No events to listen to, please choose at least one.");
+      }
+      console.log("Generating signature key...");
+      const uuidKey = uuidv4();
+      this._setSignatureKey(uuidKey);
+
+      console.log("Getting user info...");
+      const userInfo = await this.app.getUserInfo();
+
+      console.log("Registering webhook...");
+      const webhook = await this.app.createWebhookSubscription(
+        this.events,
+        this.http.endpoint,
+        userInfo.resource.current_organization,
+        userInfo.resource.uri,
+        uuidKey,
+      );
+
+      this._setWebhook(webhook.resource);
+      console.log("Webhook registered!");
+    },
+    _getCalendlySignature(event) {
+      const calendlySignature = event.headers["calendly-webhook-signature"];
+      const {
+        t,
+        signature,
+      } = calendlySignature.split(",").reduce((acc, currentValue) => {
+        const [
+          key,
+          value,
+        ] = currentValue.split("=");
+
+        if (key === "t") {
+          acc.t = value;
+        }
+
+        if (key === "v1") {
+          acc.signature = value;
+        }
+
+        return acc;
+      }, {
+        t: "",
+        signature: "",
+      });
+      return {
+        t,
+        signature,
+      };
+    },
+    _checkHmac(t, hmac, bodyRaw) {
+      console.log("Checking HMAC...");
+      const signatureKey = this._getSignatureKey();
+      const data = `${t}.${bodyRaw}`;
+      const expectedSignature = crypto.createHmac("sha256", signatureKey)
+        .update(data, "utf8")
+        .digest("hex");
+
+      if (hmac !== expectedSignature) {
+        throw new Error("Invalid HMAC Signature, connection aborted.");
+      }
+    },
+    _checkReplayAttack(t) {
+      console.log("Checking for Replay Attack...");
+      const threeMinutes = 180000;
+      const tolerance = threeMinutes;
+      const timestampMilliseconds = Number(t) * 1000;
+
+      if (timestampMilliseconds < Date.now() - tolerance) {
+        throw new Error("Invalid Timestamp Signature. The signature's timestamp is outside of the tolerance zone.");
+      }
+    },
+  },
+  hooks: {
+    async activate() {
+      const previousWebhook = this._getWebhook();
+      if (previousWebhook) {
+        await this._deactivateWebhook(previousWebhook);
+      }
+
+      await this._registerWebhook();
+    },
+  },
```

Comment:
Don't we need a `deactivate` function here? How is deactivated webhook deleted from calendly? Please excuse me if I miss something here.

Reply:
Oops!! I missed it